src/widget/weffectchainpresetbutton.cpp

```diff
@@ -0,0 +1,124 @@
+#include "widget/weffectchainpresetbutton.h"
+
+#include <QCheckBox>
+#include <QWidgetAction>
+
+#include "effects/presets/effectpresetmanager.h"
+#include "widget/effectwidgetutils.h"
+
+WEffectChainPresetButton::WEffectChainPresetButton(QWidget* parent, EffectsManager* pEffectsManager)
+        : QPushButton(parent),
+          WBaseWidget(this),
+          m_pEffectsManager(pEffectsManager),
+          m_pChainPresetManager(pEffectsManager->getChainPresetManager()),
+          m_pMenu(make_parented<QMenu>(new QMenu(this))) {
+    setMenu(m_pMenu.get());
+    connect(m_pChainPresetManager.data(),
+            &EffectChainPresetManager::effectChainPresetListUpdated,
+            this,
+            &WEffectChainPresetButton::populateMenu);
+    setFocusPolicy(Qt::NoFocus);
+}
+
+void WEffectChainPresetButton::setup(const QDomNode& node, const SkinContext& context) {
+    m_iChainNumber = EffectWidgetUtils::getEffectUnitNumberFromNode(node, context);
+    m_pChain = EffectWidgetUtils::getEffectChainFromNode(
+            node, context, m_pEffectsManager);
+    for (const auto& pEffectSlot : m_pChain->getEffectSlots()) {
+        connect(pEffectSlot.data(),
+                &EffectSlot::effectChanged,
+                this,
+                &WEffectChainPresetButton::populateMenu);
+        connect(pEffectSlot.data(),
+                &EffectSlot::parametersChanged,
+                this,
+                &WEffectChainPresetButton::populateMenu);
+    }
+    populateMenu();
+    // TODO: set icon
```

Comment:
```suggestion
    // TODO: enable applying external stylesheets
```
right now no styles can be applied, neither by using `WEffectChainPresetButton` nor an ObjectName.
the OS palette is used instead:
![image](https://user-images.githubusercontent.com/5934199/109448337-ff549f00-7a45-11eb-9396-740ec02fb31c.png)